device/peripherals/SHTC3/SHTC3.h

```diff
@@ -2,190 +2,284 @@
+     *
+     * @param command16 The command to write
+     * @return RetType The scheduler status
+     */
     RetType writeCommand(SHTC3_CMD command16) {
         RESUME();
+        addr.dev_addr = (SHTC3_I2C_ADDR << 1);
 
-        uint8_t command8[2] = {};
+        uint8_t command8[2];
         uint16ToUint8(command16, command8);
-
-        addr.mem_addr = command16;
-        addr.mem_addr_size = 2;
-
-        RetType ret = CALL(mI2C->write(addr, command8, 2));
-        if (ret != RET_SUCCESS) return ret;
+        RetType ret = CALL(mI2C.transmit(addr, command8, 2, 80));
+        if (ret != RET_SUCCESS)
+            return ret;
 
         RESET();
         return RET_SUCCESS;
     }
 
+    /**
+     * @brief Write a command to the sensor, and read the response
+     *
+     * @param command16 The command to write
+     * @param buff The buffer to read into
+     * @param numBytes The number of bytes to read
+     * @return RetType The scheduler status
+     */
     RetType readCommand(SHTC3_CMD command16, uint8_t *buff, uint8_t numBytes) {
         RESUME();
+        uint16ToUint8(command16, buff);
+        static uint8_t secondAddr = addr.dev_addr;
+        secondAddr |= 0x01 << 0;
 
-        static uint8_t command8[2] = {};
-        uint16ToUint8(command16, command8);
-
-        addr.mem_addr = command16;
-        addr.mem_addr_size = 2;
-
-
-        RetType ret = CALL(mI2C->write(addr, command8, 2));
-        if (ret != RET_SUCCESS) return ret;
-
-        ret = CALL(mI2C->read(addr, buff, numBytes));
-        if (ret != RET_SUCCESS) return ret;
+        RetType ret = CALL(mI2C.transmitReceive(addr, buff, 2, numBytes, 50, secondAddr));
+        if (ret != RET_SUCCESS)
+            return ret;
 
         RESET();
         return RET_SUCCESS;
     }
 
+    /**
+     * @brief Read the ID of the sensor
+     *
+     * @param id A pointer to the ID variable
+     * @return RetType The scheduler status
+     */
     RetType getID(uint16_t *id) {
         RESUME();
 
-        static uint8_t data[3] = {};
-        RetType ret = CALL(readCommand(READ_ID_CMD, data, 3));
-        if (ret != RET_SUCCESS) return ret;
+        static uint8_t data[2] = {};
+        RetType ret = CALL(readCommand(READ_ID_CMD, data, 2));
+        if (ret != RET_SUCCESS)
+            return ret;
 
-        *id = data[0];
-        *id <<= 8;
-        *id |= data[1];
+        *id = data[0] << 8 | data[1];
 
         RESET();
         return RET_SUCCESS;
     }
 
-    RetType setPowerMode(bool lowPowerMode) {
-        this->inLowPowerMode = lowPowerMode;
-    }
-
+    //    RetType setPowerMode(bool lowPowerMode) {
+    //        this->inLowPowerMode = lowPowerMode;
+    //    }
 
-private:
-    I2CDevice *mI2C;
+   private:
+    /* The I2C object */
+    I2CDevice &mI2C;
+    /* The I2C address of the sensor */
     I2CAddr_t addr;
+    /* Is the sensor in low power mode */
     bool inLowPowerMode;
-
+    /* The ID of the sensor */
+    uint16_t id;
```

Comment:
Maybe don't need the id as a field. You could probably just make this a static variable inside the init function

Reply:
It works right now. Ima leave it as is.